Connected System Plug-in (CSP) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/googledrive/templates/GoogleDriveListFilesIntegrationTemplate.java

```diff
@@ -0,0 +1,98 @@
+import java.util.Map;
+
+import com.appian.connectedsystems.simplified.sdk.SimpleIntegrationTemplate;
+import com.appian.connectedsystems.simplified.sdk.configuration.SimpleConfiguration;
+import com.appian.connectedsystems.templateframework.sdk.ExecutionContext;
+import com.appian.connectedsystems.templateframework.sdk.IntegrationResponse;
+import com.appian.connectedsystems.templateframework.sdk.TemplateId;
+import com.appian.connectedsystems.templateframework.sdk.configuration.PropertyPath;
+import com.appian.connectedsystems.templateframework.sdk.diagnostics.IntegrationDesignerDiagnostic;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.model.File;
+import com.google.api.services.drive.model.FileList;
+import com.google.common.base.Stopwatch;
+
+@TemplateId(name = "GoogleDriveListFilesIntegrationTemplate")
+public class GoogleDriveListFilesIntegrationTemplate extends SimpleIntegrationTemplate {
+  private static final String FOLDER_ID_KEY = "folderId";
+
+  @Override
+  protected SimpleConfiguration getConfiguration(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      PropertyPath updatedProperty,
+      ExecutionContext executionContext) {
+    return integrationConfiguration.setProperties(
+        textProperty(FOLDER_ID_KEY).label("Folder Id")
+            .instructionText("If left blank, it will list all existing files. Otherwise it will list the " +
+                "children files in the folder")
+            .build()
+    );
+  }
+
+  @Override
+  protected IntegrationResponse execute(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      ExecutionContext executionContext) {
+    String folderId = integrationConfiguration.getValue(FOLDER_ID_KEY);
+
+    IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder = IntegrationDesignerDiagnostic
+        .builder();
+    Stopwatch stopwatch = Stopwatch.createStarted();
+
+    //GoogleCredential is used to create a Google client with Drive. To send a file, you need to supply a File
+    //and an InputStreamContent.
+    GoogleCredential credential = new GoogleCredential().setAccessToken(
+        executionContext.getAccessToken().get());
+    Drive drive = new Drive.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance(),
+        credential).build();
+
+    ArrayList<File> queriedFiles = new ArrayList<File>();
+    Drive.Files.List request;
+    try {
+      request = drive.files().list();
+      if (folderId != null && !folderId.isEmpty()) {
+        request.setQ("'" + folderId + "' in parents");
+      }
+      do {
```

Comment:
Not sure we should encourage this behavior. At least put a note saying that in a production use case, we'd recommend retaining paging.

Reply:
that makes sense. I'll just remove it and take whatever is given in the first request. it just clutters this example